libs/ml/include/ml/meta/ml_type_traits.hpp

```diff
@@ -42,6 +47,13 @@ constexpr bool IsNotTrainable = !IsTrainable<T, OperationType>;
 template <typename T, typename OperationType>
 constexpr bool IsGraph = std::is_base_of<fetch::ml::Graph<T>, OperationType>::value;
 
+template <typename T, typename OperationType>
+constexpr bool IsShareable =
```

Comment:
please raise an issue that we should implement shareable weights in any other relevant ops/layers